app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
+  validates :name, :surname, presence: true
+
+  validate :password_validation
+
+  private
+
+  def password_validation
+    if !password.match(/[A-Z]{1}/) || !password.match(/[a-z]{1}/) || !password.match(/[0-9]{1}/)
```

Comment:
I struggled to understand this line, do you think we could make a bit easier by wrapping each check into its own method? Something like this:

```ruby
if !has_uppercase?(password) || !has_lowercase?(password) || !has_number?(password)
```

We could even have it read as more natural english if we did something like this instead:

```ruby
if !(has_uppercase?(password) && has_lowercase?(password) && has_number?(password))
```

wdyt?